Build script changes

alpaix opened the pull request:
commit e81a849f4796afd37910b6ba56c5c3b50d18b063
Author: Alex Panov alpaix@outlook.com
Date:   Tue Jan 19 16:51:30 2016 -0800

```
Reverted combined dnu pack to separate per-project pack's

Fixes https://github.com/NuGet/Home/issues/1957.
+ Corrected pipeline input processing
```

commit 11d507bf2e9dbb7d2d8ecb884edff5b268a9b116
Author: Alex Panov alpaix@outlook.com
Date:   Wed Jan 13 23:36:47 2016 -0800

```
Fixed ILMerge function

Added query to feed all dll from build folder
```

commit 385308791509e08feb572ed93e2132543160f7b0
Author: Alex Panov alpaix@outlook.com
Date:   Wed Jan 13 22:50:55 2016 -0800

```
Combined delay sign parameters

Shortened arguments list of a build script
```

commit 202dca10d7694255a7d3556567de0112ac7531fe
Author: Alex Panov alpaix@outlook.com
Date:   Wed Jan 13 22:27:35 2016 -0800

```
Removed redundant parameters in build script

Default cmdlet parameters passed by default.
```

commit 5219ba8e0492822cf574a389daab9bcf3f526cc0
Author: Alex Panov alpaix@outlook.com
Date:   Wed Jan 13 22:11:26 2016 -0800

```
Read package sources from NuGet.Config

Modified build script to always read sources from the config file
```

commit 72fc041500797bc0f53c2440dbabb871a5fc5dd7
Author: Alex Panov alpaix@outlook.com
Date:   Wed Jan 13 21:04:43 2016 -0800

```
Revisiting Verbose parameter impact at the build script

- Decoupled Verbose from Fast (now it's possible to build fast WITH
  verbose output)
- Turned functions into Cmdlets where it's necessary to propogate Verbose
  value
- Switched dnx test reporter based on Verbose preference value
```

alpaix committed:
Revisiting Verbose parameter impact at the build script

- Decoupled Verbose from Fast (now it's possible to build fast WITH
  verbose output)
- Turned functions into Cmdlets where it's necessary to propogate Verbose
  value
- Switched dnx test reporter based on Verbose preference value

alpaix committed:
Read package sources from NuGet.Config

Modified build script to always read sources from the config file

alpaix committed:
Removed redundant parameters in build script

Default cmdlet parameters passed by default.

alpaix committed:
Combined delay sign parameters

Shortened arguments list of a build script

alpaix committed:
Fixed ILMerge function

Added query to feed all dll from build folder

alpaix committed:
Reverted combined dnu pack to separate per-project pack's

Fixes https://github.com/NuGet/Home/issues/1957.
+ Corrected pipeline input processing

alpaix commented:
Fixes https://github.com/NuGet/Home/issues/1957.

//cc @emgarten @yishaigalatzer @deepakaravindr

emgarten commented on build/common.ps1:
```diff
@@ -492,10 +573,23 @@ Function Invoke-ILMerge {
     )
     $nugetArtifactsFolder = Join-Path $Artifacts "$Configuration\NuGet.Clients\NuGet.CommandLine"
-    pushd $nugetArtifactsFolder
+    $dlls = Get-ChildItem $nugetArtifactsFolder -Filter 'nuget.*.dll' | %{ $_.Name }
```
This is nice, but aren't some dlls like xplat getting in now? The CI is I think using a different script for this, so for the official build they are likely excluded and this can still work.

emgarten replied:
Do the Visual Studio dlls show up also?

yishaigalatzer replied:
I agree with @emgarten here, we should probably keep the whitelist. The way I have seen done is to have a whitelist+blacklist and fail if a new dll shows up

yishaigalatzer replied:
here is a suggestion:
1. Create a whitelist and blacklist, they can be two files or one, they can be json or two simple flat files with each dll name on a line
2. When processing make sure each dll in the whitelist exists, or else fail with an error
3. make sure each dll exists either in the whitelist or the blacklist or else fail with an error
4. make sure each dll in the black list exists or warn if it is missing
5. Errors and warnings should point to the files to be edited explicitly
6. The CI process should use exactly the same files as the build script
7. This change should also apply to the vsix include/exclude process

alpaix replied:
@emgarten It collects dlls from the project's output folder NuGet.CommandLine therefore only relevant assemblies produced by build are in.

alpaix replied:
@yishaigalatzer The intention was to have a one-liner with no need to update it every time dll structure changes.
But I understand the idea of controlling dependencies. Will revisit the method as you suggested.

emgarten commented:
Removing the ilmerge white list worries me, but the rest looks good.

alpaix committed:
Enhanced ilmerging function with include/ignore file lists

- Added ability to control included/ignored artifacts thru .mergeinclude,
  .mergeignore lists of files

emgarten commented:
:shipit:
